include/big_core_input.h

```diff
@@ -93,21 +96,27 @@ class BigCoreInput : public BigCoreBase
     //template<typename T>
     //std::vector<T> getTile(uint64_t imageNum, uint64_t tileNum = 0);
     // RV: Libore, ve Tve verzi bych musel vzdy zadavat datovy typ T, takto ho prekladac odvodi sam ze vstupne-vystupniho vektoru, viz big_core_test.cpp
+	// LB: v obou pripadech musi znat datovy typ, jaky chce, taky nevidim rozdil, akorat sev tomhle pripade pri volani mene pise :)
     template<typename T>
     void getTile(std::vector<T>& data, uint64_t imageNum, uint64_t tileNum = 0);
 
     // todo: introduce constant iterators to iterate through tiles from first tile of the first image to last tile of the last image.
 
 protected:
 
+	// LB: jestli tahle metoda ma byt protected, tak je zbytecne, aby byla staticka
     static bool checkHeader(const char* buffer);
     bool readChunk(std::ifstream& file, uint64_t& id, uint64_t& length);
     void checkDataIntegrity(uint64_t& thisData, const uint64_t loadedData);
     bool readData(std::ifstream& file, const uint64_t id, const uint64_t length);
 
 protected:
     // RV: lock mi prijde zbytecny. Pokud je soubor spatne vytvoren, proste to bude fungovat divne, ale stejne ho neopravime.
+	// LB: ja to myslel jinak, ve chvili kdy vytvarim instanci tehle tridy, muzu menit parametry,
+	// LB: ve chvili, kdy prdavam data z dalsiho souboru (aspon tak jsem pochopil, ze to ma fungovat), uz nemuzu menit parametry a zaroven ty parametry musi odpovidat puvodnim
+	// LB: tj. lock = true, viz readData a checkDataIntegrity
```

Comment:
U BigCoreInput nelze nijak menit parametry, proto je lock zbytecny. Naopak u BigCoreOutput bude neco jako lock samozrejme potreba.

Reply:
Ano, viz vyse. Instance vlastne slouzi k tomu, abych si z ni tahal na prani data, ktera zrovna potrebuji. Kdyz chci ale data z jineho souboru, musim si vytvorit dalsi instanci (nebo stavajici soubor zavrit a otevrit novy).

Typicke pouziti bude renderovani BTF textur. Kdyz mas nejakou 3D scenu, kazdy objekt ma vlastni material. Pri pouziti klasickych textur tedy nahrajes do pameti graficke karty obrazky vsech textur a shader si z nich vyzobava prave potrebna data. U BTF textur se jedna textura sklada z velkeho mnozstvi obrazku (ty reprezentuji jak material vypada pro ruzna osvetleni, kdyz se na nej divame z ruznych smeru). Renderovani (z nekomprimovanych, tj. nedegradovanych) dat z principu nebude v realnem case, ale stejne chceme, aby bylo co nejrychlejsi. Pro kazdou BTF texturu si tedy vytvorime jednu instanci objektu BigCoreInput a renderer si naprikla operatorem ( ) muze vyzobavat potrebna data. Pri otevreni souboru mi tedy nevadi, kdyz to chvilku trva, protoze se to provede pouze jednou pro kazdou texturu. Tahani dat pak ale musi byt maximalne rychle, protoze se bude opakovat milionkrat.

Podobne to bude u BigCoreOutput. Vytvorim instanci, nastavim rozmery, dataOrder, datove typy a pak muzu v libovolnem poradi posilat jakakoli data. Ta se budou ukladat do pameti a v destruktoru, pri zavreni souboru nebo pri otevreni jineho souboru se to zapise na disk do souboru. Pripadne tam bude rezim, ze se na disk zapisuje okamzite a na konci se soubor pouze zavre. Ale vzdy jedna instance pracuje jen s jednim souborem.